### PR TITLE
[PEE-20] Add dummy protected routes

### PR DESCRIPTION
**Pull request by seungriyou**
## 💫 Motivation

- protected routes for logged in user and email-verified user are required

<br>

## 📌 Key Changes

- add `get_user_by_id()` to `UserService`
- add two global dependencies for protected routes
    - `get_current_user`: get current user and return it
    - `get_current_verified_user`: get current user, return it if the user is verified and raise `UserNotVerifiedError` if the user is not verified
- create two dummy protected endpoints for them

<br>

## 📢 To Reviewers

- Is it appropriate to set `tokenUrl="/api/v1/login"` in `core.dependencies.auth`?  I don't know if there's a better way...
- I found that better convention for endpoint url is hypen(`-`) than underscore(`_`). I think it would be great to specify our convention for implementing API. (Currently, endpoint url for refresh token is `/refresh_token`)
    
    > ref:
    > 
    > - https://stackoverflow.com/questions/119312/urls-dash-vs-underscore
    > - [https://velog.io/@caesars000/RESTful-API-가이드](https://velog.io/@caesars000/RESTful-API-%EA%B0%80%EC%9D%B4%EB%93%9C)

### CODE REVIEW

File: peerloop/domain/user/views.py [hunk 1]
```diff
@@ -49,3 +52,15 @@ async def verify_email(
...
+@router.get("/me", status_code=status.HTTP_200_OK, response_model=UserResponse)
+@inject
+async def get_me(user: Annotated[User, Depends(get_current_user)]) -> User:
+    return user
+
+
+@router.get("/verified-me", status_code=status.HTTP_200_OK, response_model=UserResponse)
+@inject
+async def get_verified_me(user: Annotated[User, Depends(get_current_verified_user)]) -> User:
+    return user
```
Comment (noisrucer) [inline peerloop/domain/user/views.py:66]: Why do we need these two routes? I wonder if they're for testing purposes.

Comment (noisrucer) [inline peerloop/domain/user/views.py:66]: Why do we need these two routes? I wonder if they're for testing purposes.

File: peerloop/core/dependencies/auth.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    user_id = user_service.token_manager.decode_token(token=token)
+    user = await user_service.get_user_by_id(user_id=int(user_id))
+    return user
+
+
+@inject
+async def get_current_verified_user(verified_user: Annotated[User, Depends(get_current_user)]) -> User:
+    if not verified_user.is_verified:
+        raise UserNotVerifiedError("User Not Verified Error: email verification is required.")
+    return verified_user
```
Comment (noisrucer) [inline peerloop/core/dependencies/auth.py:28]: What do you think is more appropriate between returning the user model and returning the user id?

I suppose we generally only need `user_id` as a primary key to interact with the database but I'm not sure what's the best practice.

**Review by noisrucer — COMMENTED** [review]
_

**Comment by noisrucer** [comment]
> ## 💫 Motivation
> * protected routes for logged in user and email-verified user are required
> 
> 
> ## 📌 Key Changes
> * add `get_user_by_id()` to `UserService`
> * add two global dependencies for protected routes
>   
>   * `get_current_user`: get current user and return it
>   * `get_current_verified_user`: get current user, return it if the user is verified and raise `UserNotVerifiedError` if the user is not verified
> * create two dummy protected endpoints for them
> 
> 
> ## 📢 To Reviewers
> * Is it appropriate to set `tokenUrl="/api/v1/login"` in `core.dependencies.auth`?  I don't know if there's a better way...
> * I found that better convention for endpoint url is hypen(`-`) than underscore(`_`). I think it would be great to specify our convention for implementing API. (Currently, endpoint url for refresh token is `/refresh_token`)
>   > ref:
>   > 
>   > * https://stackoverflow.com/questions/119312/urls-dash-vs-underscore
>   > * [https://velog.io/@caesars000/RESTful-API-가이드](https://velog.io/@caesars000/RESTful-API-%EA%B0%80%EC%9D%B4%EB%93%9C)

* I didn't think of another way. Why not?!
* Thanks for the information! I didn't know that hypen delimiter is a standard practice and in fact it was. Could you change the endpoint name to `refresh-token`?

File: peerloop/domain/user/views.py (hunk 1 above)
Comment (seungriyou) [reply peerloop/domain/user/views.py:66]: Yes, I made them in order to test `get_current_user` and `get_current_verified_user` in swagger, so these endpoints are unnecessary except for test purposes!

`/me` endpoint is about "log in" and `/verified-me` endpoint is about "log in" & "email verification". 

Would it be appropriate to remove these routes?

File: peerloop/core/dependencies/auth.py (hunk 2 above)
Comment (seungriyou) [reply peerloop/core/dependencies/auth.py:28]: I was thinking about it, too. For now, I implemented these functions according to FastAPI document tutorial, which looks good in that it is possible to get basic information about users without additional database access. Most of the time we're going to use this function is when we need basic information about that user!

But I don't know what's the best practice, too...! 😭

**Comment by seungriyou** [comment]
> * I didn't think of another way. Why not?!
> * Thanks for the information! I didn't know that hypen delimiter is a standard practice and in fact it was. Could you change the endpoint name to `refresh-token`?

@noisrucer
- I was just worried because it felt like a specific url `/api/v1/login` was hardcoded in the code. If the api version changes from `v1` to `v2`, I wonder if I need to change url directly....
- Sure!! 😆

**Review by noisrucer — APPROVED** [review]
LGTM!

**Review by noisrucer — APPROVED** [review]
LVGTM

**Pull request merged**